middleware/logging.go

```diff
@@ -58,7 +58,7 @@ func (l Log) logWithRequest(r *http.Request) log.Logger {
 	localLog := l.Log
 	traceID, ok := tracing.ExtractTraceID(r.Context())
 	if ok {
-		localLog = log.With(localLog, "traceID", traceID)
+		localLog = log.With(localLog, "trace_id", traceID)
```

Comment:
This is logging-only; is it better to be consistent with OpenTelemetry here, or consistent with what we used to do?

Reply:
Internally, our Loki datasources are configured to support either. I think making it consistent is better but I don't feel strongly about it.

Reply:
TIL, thanks.  Since this is a public open source project we shouldn't be particularly guided by what happens inside Grafana Labs, but I'll take comfort that anyone can apply the same setting.
<img width="482" alt="image" src="https://github.com/grafana/dskit/assets/8125524/d8663ad7-d1d5-488d-b10c-bc6f273dcda1">